src/shared/components/case-viewer/case-viewer.component.ts

```diff
@@ -116,33 +120,27 @@ export class CaseViewerComponent implements OnInit, OnDestroy {
         if (result === 'Delete') {
           this.draftService.deleteDraft(this.caseDetails.case_id)
             .subscribe(_ => {
-              return this.router.navigate(['list/case'])
-                .then(() => {
-                  this.alertService.setPreserveAlerts(true);
-                  this.alertService.success(`The draft has been successfully deleted`);
-                });
+              this.navigationNotifierService.announceNavigation({action: NavigationOrigin.DRAFT_DELETED});
             }, _ => {
-              return this.router.navigate(['list/case']);
+              this.navigationNotifierService.announceNavigation({action: NavigationOrigin.ERROR_DELETING_DRAFT});
             });
         }
       });
     } else if (this.isDraft() && trigger.id !== CaseViewTrigger.DELETE) {
       theQueryParams[DRAFT_QUERY_PARAM] = this.caseDetails.case_id;
       theQueryParams[CaseViewerComponent.ORIGIN_QUERY_PARAM] = 'viewDraft';
-      return this.router.navigate(
-        ['create/case',
-          this.caseDetails.case_type.jurisdiction.id,
-          this.caseDetails.case_type.id,
-          trigger.id], {queryParams: theQueryParams}).catch(error => {
-        this.handleError(error, trigger)
```

Comment:
We have to make sure we handle errors in the consumer appropriately.

Reply:
done on the subscriber side